ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -0,0 +1,118 @@
+import { PRIMARY, SECONDARY } from '../helpers/constants/common'
+import { getTokenToAddress } from '../helpers/utils/token-util'
+import { useUserPreferencedCurrency } from './useUserPreferencedCurrency'
+import { formatDateWithYearContext, shortenAddress } from '../helpers/utils/util'
+import {
+  APPROVED_STATUS,
+  SUBMITTED_STATUS,
+} from '../helpers/constants/transactions'
+import { useCurrencyDisplay } from './useCurrencyDisplay'
+import { useTokenDisplayValue } from './useTokenDisplayValue'
+import { useTokenData } from './useTokenData'
+import { tokenSelector } from '../selectors'
+
+const SEND = 'send'
+const DEPOSIT = 'deposit'
+const INTERACTION = 'interaction'
+const APPROVAL = 'approval'
+
+// This is duplicated from transactions selectors, additionally
+// from the history view we will already know the pending status
+// of a transaction, but in order to make this hook less coupled
+// to where it is used we should determine the pending status here
+const pendingStatusMap = {
+  [APPROVED_STATUS]: true,
+  [SUBMITTED_STATUS]: true,
+}
+
+/**
+ * @typedef {Object} TransactionDisplayData
+ * @property {string} title               - primary description of the transaction
+ * @property {string} subtitle            - supporting text describing the transaction
+ * @property {string} category            - the transaction category
+ * @property {string} primaryCurrency     - the currency string to display in the primary position
+ * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
+ * @property {string} status              - the status of the transaction
+ * @property {string} senderAddress       - the Ethereum address of the sender
+ * @property {string} recipientAddress    - the Ethereum address of the recipient
+ */
+
+/**
+ * useTransactionDisplayData
+ *
+ * The goal of this method is to perform all of the necessary computation and
+ * state access required to take a transactionGroup and derive from it a shape
+ * of data that can power all views related to a transaction. Presently the main
+ * case is for shared logic between transaction-list-item and transaction-detail-view
+ * @param {Object} transactionGroup group of transactions
+ * @return {TransactionDisplayData}
+ */
+export function useTransactionDisplayData (transactionGroup) {
+  const knownTokens = useSelector(tokenSelector)
+  const t = useI18nContext()
+  const { initialTransaction, primaryTransaction } = transactionGroup
+  // initialTransaction contains the data we need to derive the primary purpose of this transaction group
+  const { transactionCategory } = initialTransaction
+
+  const { from: senderAddress, to } = initialTransaction.txParams || {}
+
+  // for smart contract interactions, methodData can be used to derive the name of the action being taken
+  const methodData = useSelector((state) => getKnownMethodData(state, primaryTransaction?.txParams?.data)) || {}
```

Comment:
Previously this had used the initial transaction, not the primary transaction. `mergeProps` in `transaction-list-item.container.js` passes the initial transaction as `transaction`, which is passed through to `transaction-action.component.js`.
```suggestion
  const methodData = useSelector((state) => getKnownMethodData(state, initialTransaction?.txParams?.data)) || {}
```

Reply:
..... 🤦‍♂️ Maybe not.

It looks like the old `TransactionAction` component had taken the `methodData` of the primary transaction, but had used the initial transaction when looking for the action key. Oops!

... I'm having trouble understanding why that would be desirable though.

Reply:
Apparently this was broken in #6551 . It was supposed to use the initial transaction, as it did before that PR.

Reply:
![200w_d](https://user-images.githubusercontent.com/4448075/82695591-0040e080-9c2b-11ea-8ace-95671835506c.gif)

Reply:
so your suggestion is ... valid still?

Reply:
Yeah I think so 😅